### PR TITLE
configure virt-handler max-device with kubelet maxPods

### LINKED ISSUE #6082: [Sig-scale] enable virt-operator to configure virt-handler to run more than 110 VMs per node
/kind enhancement

**Motivation**:
In the recent [SIG-scale meeting](https://docs.google.com/document/d/1d_b2o05FfBG37VwlC2Z1ZArnT9-_AEJoQTe7iKaQZ6I/edit#heading=h.50eclt5fctl9), we are discussing to run a scale test with more than 110 VMIs per node. It is trivial to increase the max number of pods per node, but the max number of VMIs is not straightforward. 

**What happened**:
Currently, `virt-handler` already have `max-devices` flag to increase the number of VMs per node
https://github.com/kubevirt/kubevirt/blob/0514a664151863d901d90694feb08c86df0076bb/cmd/virt-handler/virt-handler.go#L527

However, the `virt-handler` configuration is determined by the `virt-operator` that reverts all changes in the `virt-handler` Daemonset during the reconcile (of course, when we deploy it using the operator) .

**What you expected to happen**:
I expect to be able to configure `virt-operator` to create all `virt-handler` with a given max-devices number. Currently, all the `virt-handler` Daemonset options in `virt-operator` are hardcoded:
https://github.com/kubevirt/kubevirt/blob/0514a664151863d901d90694feb08c86df0076bb/pkg/virt-operator/resource/generate/components/daemonsets.go#L105

### PR DESCRIPTION
**Pull request by marceloamaral**
**What this PR does / why we need it**:
In the recent [SIG-scale meeting](https://docs.google.com/document/d/1d_b2o05FfBG37VwlC2Z1ZArnT9-_AEJoQTe7iKaQZ6I/edit#heading=h.50eclt5fctl9), we are discussing to run a scale test with more than 110 VMIs per node. It is trivial to increase the max number of pods per node, but the max number of VMIs is not straightforward. 

Currently, `virt-handler` already have `max-devices` flag to increase the number of VMs per node
https://github.com/kubevirt/kubevirt/blob/0514a664151863d901d90694feb08c86df0076bb/cmd/virt-handler/virt-handler.go#L527 However, the `virt-handler` configuration is determined by the `virt-operator` that reverts all changes in the `virt-handler` Daemonset during the  strict-reconciliation.

This PR change `virt-handler` to detect the per node pod limit and use that as our VMI limit (`max-devices`), instead of using the hardcoded 110. 

This PR is related to #6082

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Signed-off-by: Marcelo Amaral <marcelo.amaral1@ibm.com>

### CODE REVIEW

File: cmd/virt-handler/virt-handler.go [hunk 1]
```diff
@@ -312,6 +315,20 @@ func (app *virtHandlerApp) Run() {
...
+		yamlFile, err := ioutil.ReadFile(kubeletConfigDir)
+		if err != nil {
+			glog.Fatalf("Error reading kubelet config: %v", err)
+		}
+		var data map[string]interface{}
+		err = yaml.Unmarshal(yamlFile, &data)
+		if err != nil {
+			glog.Fatalf("Error unmarshalling kubelet config: %v", err)
+		}
+		app.MaxDevices = data["maxPods"].(int)
```
Comment (ashleyschuett) [inline cmd/virt-handler/virt-handler.go:328]: I don't think this value is required or defaulted in the configuration file. We should set a default if this is not present.

**Comment by ashleyschuett** [comment]
I also created a PR for this before seeing yours https://github.com/kubevirt/kubevirt/pull/6105 I removed the kubelet configuration logic in the last commit so it just takes into account if the user wants to set a different value than default/whats set for kubelet

**Comment by marceloamaral** [comment]
@ashleyschuett we have discussed In the recent SIG-scale meeting, and @rmohr suggested to use a default high value, like 1000 devices. Instead of find the value on the kubelet-config.yml file since that path might vary

File: cmd/virt-handler/virt-handler.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ const (
 	podIpAddress = ""
 
 	// This value is derived from default MaxPods in Kubelet Config
```
Comment (zcahana) [inline cmd/virt-handler/virt-handler.go:100]: Following the change to 1000, the above comment is no longer relevant.

**Review by zcahana — COMMENTED** [review]
I didn't see the initial implementation in this PR, but from the comments I understand that this was initially implemented by looking up the maxPods in the kubelet config file. I agree that this is somewhat fragile (the file location can change, the `--max-pods` flag can be used despite being deprecated). Here's another approach which seems to me as more robust: how about just reading that value from the Node object (`.status.capacity.pods`)? That value gets updated by the kubelet, regardless of how that config came in, and is guaranteed to be set before the virt-handler pod is started.

And another small nit: the PR description says _fixes #6082_ although it doesn't really (I believe @ashleyschuett's #6105 does). It's probably better to remove this to avoid auto-closing the issue when this is merged.

File: cmd/virt-handler/virt-handler.go (hunk 2 above)
Comment (marceloamaral) [reply cmd/virt-handler/virt-handler.go]: Updated, thanks!

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

**Comment by rmohr** [comment]
@marceloamaral could you update the PR description? Just to avoid confusion on what we have in kubevirt since the description does not match the content anymore.

**Pull request merged**